src/parquet/arrow/writer.cc

```diff
@@ -497,7 +497,7 @@ Status FileWriter::Impl::TypedWriteBatch<BooleanType, ::arrow::BooleanType>(
   auto writer = reinterpret_cast<TypedColumnWriter<BooleanType>*>(column_writer);
 
   int buffer_idx = 0;
-  int32_t offset = array->offset();
+  int32_t offset = static_cast<int32_t>(array->offset());
```

Comment:
Let's use `int64_t` here instead. In the instances above we should maybe use int64 also, @xhochy what do you think?